File: 01-basic-api/examples/hello_claude.js
#!/usr/bin/env node
/**
 * Your first Claude API call!
 *
 * This script demonstrates:
 * - Loading environment variables
 * - Creating an Anthropic client
 * - Making a simple API call
 * - Displaying the response and token usage
 *
 * Usage:
 *     node hello_claude.js
 */

import Anthropic from '@anthropic-ai/sdk';
import dotenv from 'dotenv';

// Load environment variables from .env file
dotenv.config();

// Initialize the Anthropic client
const client = new Anthropic({
  apiKey: process.env.ANTHROPIC_API_KEY
});

async function main() {
  console.log('🤖 Making your first Claude API call...\n');

  // Create a message
  const message = await client.messages.create({
    model: 'claude-3-5-haiku-20241022',
    max_tokens: 1024,
    messages: [
      {
        role: 'user',
        content: 'Say hello and introduce yourself in one sentence!'
      }
    ]
  });

  // Extract the response text
  const responseText = message.content[0].text;

  // Display results
  console.log('='.repeat(60));
  console.log('Claude\'s Response:');
  console.log('='.repeat(60));
  console.log(responseText);
  console.log('\n' + '='.repeat(60));
  console.log('📊 Token Usage:');
  console.log('='.repeat(60));
  console.log(`  Input tokens:  ${message.usage.input_tokens}`);
  console.log(`  Output tokens: ${message.usage.output_tokens}`);
  console.log(`  Total tokens:  ${message.usage.input_tokens + message.usage.output_tokens}`);

  // Calculate approximate cost (Haiku pricing as of 2025)
  const inputCost = (message.usage.input_tokens / 1_000_000) * 0.25;
  const outputCost = (message.usage.output_tokens / 1_000_000) * 1.25;
  const totalCost = inputCost + outputCost;

  console.log(`\n💰 Approximate cost: $${totalCost.toFixed(6)}`);
  console.log('='.repeat(60));
}

main().catch(error => {
  console.error('Error:', error.message);
  process.exit(1);
});

File: 01-basic-api/examples/hello_claude.py
#!/usr/bin/env python3
"""
Your first Claude API call!

This script demonstrates:
- Loading environment variables
- Creating an Anthropic client
- Making a simple API call
- Displaying the response and token usage

Usage:
    python hello_claude.py
"""

import os
from anthropic import Anthropic
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Initialize the Anthropic client
client = Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))

def main():
    print("🤖 Making your first Claude API call...\n")

    # Create a message
    message = client.messages.create(
        model="claude-3-5-haiku-20241022",
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": "Say hello and introduce yourself in one sentence!"
            }
        ]
    )

    # Extract the response text
    response_text = message.content[0].text

    # Display results
    print("=" * 60)
    print("Claude's Response:")
    print("=" * 60)
    print(response_text)
    print("\n" + "=" * 60)
    print("📊 Token Usage:")
    print("=" * 60)
    print(f"  Input tokens:  {message.usage.input_tokens}")
    print(f"  Output tokens: {message.usage.output_tokens}")
    print(f"  Total tokens:  {message.usage.input_tokens + message.usage.output_tokens}")

    # Calculate approximate cost (Haiku pricing as of 2025)
    input_cost = (message.usage.input_tokens / 1_000_000) * 0.25
    output_cost = (message.usage.output_tokens / 1_000_000) * 1.25
    total_cost = input_cost + output_cost

    print(f"\n💰 Approximate cost: ${total_cost:.6f}")
    print("=" * 60)

if __name__ == "__main__":
    main()
